operator/v1/types_dns.go

```diff
@@ -86,19 +86,40 @@ type Server struct {
 	ForwardPlugin ForwardPlugin `json:"forwardPlugin"`
 }
 
+// ForwardingPolicy is the policy to use when forwarding DNS requests.
+// +kubebuilder:validation:Enum=Random;RoundRobin;Sequential
+type ForwardingPolicy string
+
+const (
+	// RandomForwardingPolicy picks a random upstream server to query
+	RandomForwardingPolicy ForwardingPolicy = "Random"
+
+	// RoundRobinForwardingPolicy picks upstream servers in a round-robin order
+	RoundRobinForwardingPolicy ForwardingPolicy = "RoundRobin"
+
+	// SequentialForwardingPolicy tries querying upstream servers in a sequential order
+	SequentialForwardingPolicy ForwardingPolicy = "Sequential"
+)
+
 // ForwardPlugin defines a schema for configuring the CoreDNS forward plugin.
 type ForwardPlugin struct {
 	// upstreams is a list of resolvers to forward name queries for subdomains of Zones.
-	// Upstreams are randomized when more than 1 upstream is specified. Each instance of
-	// CoreDNS performs health checking of Upstreams. When a healthy upstream returns an
-	// error during the exchange, another resolver is tried from Upstreams. Each upstream
-	// is represented by an IP address or IP:port if the upstream listens on a port other
-	// than 53.
+	// Each instance of CoreDNS performs health checking of Upstreams. When a healthy upstream
+	// returns an error during the exchange, another resolver is tried from Upstreams. The
+	// Upstreams are selected in the order specified in Policy. Each upstream is represented
+	// by an IP address or IP:port if the upstream listens on a port other than 53.
 	//
 	// A maximum of 15 upstreams is allowed per ForwardPlugin.
 	//
 	// +kubebuilder:validation:MaxItems=15
 	Upstreams []string `json:"upstreams"`
+
+	// policy is used to determine the order in which upstream servers are selected for querying.
+	// The default value is "Random"
```

Comment:
Generation of the CRD copies the godoc on this field into the CRD, but it doesn't copy the godoc for the type, so you need to move or repeat the information about the allowed values here.  Also, we should clarify the difference between "Sequential" and "RoundRobin"—which is a little difficult to do succinctly; maybe the following is clear and succinct enough?
```suggestion
	// policy is used to determine the order in which upstream servers are selected for querying.
	// Any one of the following values may be specified:
	//
	// * "Random" picks a random upstream server for each query.
	// * "RoundRobin" picks upstream servers in a round-robin order, moving to the next server for each new query.
	// * "Sequential" tries querying upstream servers in a sequential order until one responds, starting with the first server for each new query.
	//
	// The default value is "Random"
```